### ISSUE TITLE
AMDeviceInstallApplication failed: 0xE800003A: Unknown error.

### ISSUE THREAD

**Issue by michaKFromParis** [phonegap/ios-deploy#130]
Hi,

Thanks for such a great tool. Been using it for a while. Recently, it stopped working for our ci builds with **AMDeviceSecureInstallApplication** returning 0xE800003A

command line arguments:

```bash
ios-deploy --id c6135f55a0517572421f2101e9866501eea3c8f6 --timeout 5 --debug --noninteractive --justlaunch --bundle "build/bin/iOS-Release/SparksPlayer.app"
```

```bash
ios-deploy -V
1.6.0
```
I can deploy it successfully from Xcode. I think it would be useful to try to reverse the error codes to useful error messages. Happy to help if anyone knows where to find them but I guess **AMDeviceSecureInstallApplication** is a private API.

Thanks for your support,

M

**Comment by michaKFromParis** [comment]
Hmmmm... I just tried the demo app and I get the same result. Doesn't seem related to my app:

```bash
michaK@blue ~/Sources/ios-deploy (git::master) $ make debug
gcc -ObjC -g -o ios-deploy -framework Foundation -framework CoreFoundation -framework MobileDevice -F/System/Library/PrivateFrameworks ios-deploy.c
gcc -ObjC -g -arch armv7 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS8.3.sdk -framework CoreFoundation -o demo demo.c
mkdir -p demo.app
cp demo demo.app/
cp Info.plist ResourceRules.plist demo.app/
codesign -f -s "iPhone Developer" --entitlements Entitlements.plist demo.app
./ios-deploy --debug --bundle demo.app
[....] Waiting for iOS device to be connected
[....] Using iPhone 6 (GSM) 'iPhone 6' (c6135f55a0517572421f2101e9866501eea3c8f6) (c6135f55a0517572421f2101e9866501eea3c8f6).
------ Install phase ------
[  0%] Found iPhone 6 (GSM) 'iPhone 6' (c6135f55a0517572421f2101e9866501eea3c8f6) connected through USB, beginning install
[  5%] Copying /Users/michaK/Sources/ios-deploy/demo.app/META-INF/ to device
[  8%] Copying /Users/michaK/Sources/ios-deploy/demo.app/META-INF/com.apple.ZipMetadata.plist to device
[ 11%] Copying /Users/michaK/Sources/ios-deploy/demo.app/_CodeSignature/ to device
[ 14%] Copying /Users/michaK/Sources/ios-deploy/demo.app/_CodeSignature/CodeResources to device
[ 18%] Copying /Users/michaK/Sources/ios-deploy/demo.app/demo to device
[ 43%] Copying /Users/michaK/Sources/ios-deploy/demo.app/Info.plist to device
[ 46%] Copying /Users/michaK/Sources/ios-deploy/demo.app/ResourceRules.plist to device
[ 52%] CreatingStagingDirectory
[ 57%] ExtractingPackage
[ 60%] InspectingPackage
[ 60%] TakingInstallLock
[ 65%] PreflightingApplication
[ 70%] VerifyingApplication
AMDeviceInstallApplication failed: 0xE800003A: Unknown error.
make: *** [debug] Error 253
```

**Comment by shazron** [comment]
Most likely a provisioning profile problem. The demo app would have picked out a wildcard p.p.

You might have luck repairing with https://github.com/KrauseFx/sigh , not sure

**Comment by michaKFromParis** [comment]
Hi,

Thanks for your quick reply and for pointing out **sigh** (seems like a great initiative after fighting provisioning profiles for years).. I'll have a try when I get back to my desk.  Is there something wrong with using wildcard p. p. with ios-deploy? Might be my issue...

**Comment by shazron** [comment]
No wildcard is fine. I have a hunch a p.p/cert might have expired. Note that a p.p contains the entirety of the associated cert in itself, so if you remove an expired cert without removing an expired p.p, it will reinstall the cert.

**Comment by michaKFromParis** [comment]
Apparently, It was indeed a p. p. issue as messing around with sigh (mostly deleting all p. p. and re-installing the only one I wanted to work at this moment). But I'm quite frustrated as I can't point precisely the reason why it was failing. A fix out of nowhere as it always happens when it comes to iOS p.p. :(

The lack of textual error messages is still an issue though and I found a way to reverse the error messages from XCode though...

- build the xcodeproj in Xcode once
- regenerate the app using my build script
- Hit run in Xcode so it tries to install the regenerated .app
- See the error message

But I wonder if these error codes values change with the various versions of Xcode though.

Thanks for the help!

**Comment by shazron** [comment]
no worries, anytime! Glad that it was solved (although through "magic") ;)

**Comment by jtomaszewski** [comment]
Just a remind for others: update your ios-deploy (`sudo npm install -g ios-deploy`) if it still gives the error with no message. For me updating it to newest version helped.

**Comment by talsurizon** [comment]
hi, 
i'm getting a similar error while running using the ios-deploy command:
ios-deploy[3282:49213] [ !! ] Error 0xe8008014: unknown. AMDeviceSecureInstallApplication(0, device, url, options, install_callback, 0)

does anyone know what's wrong?..

**Comment by guillemhs** [comment]
I have the same problem. Any help?

**Comment by adudurant** [comment]
So  anyone could help？ How to solve problem？

**Comment by MikeCIDFG** [comment]
Ran into this error when trying to deploy an Ionic/Cordova app to a device from Terminal. 
I'd recently updated Cordova and uninstalled/reinstalled the iOS platform, which screwed up the provisioning profile & dev team I'd previously set in XCode. Fixing that then led me to this error.

Trying to run on a device directly from XCode gave a bit more info: 
"application’s application-identifier entitlement does not match that of the installed application."

Essentially, I had an existing version of the app already installed on the phone, which was conflicting w/the updated provisioning profile & app I was trying to install. 

Deleted the old app from the device & it finally installed from XCode, but would not launch.
I then needed to "trust" the app/developer (me) in Settings -> General -> Device Management. 
I think I could've skipped this step if I'd added the app directly in "Devices" in XCode (where I also could've downloaded the container to save any important data from the old version - in my case I didn't care).  

App now installs & runs fine on device from both XCode and Terminal.
